ExploreASL_ImportWorkflow.m

```diff
@@ -0,0 +1,115 @@
+function [x] = ExploreASL_ImportWorkflow(x)
+%ExploreASL_ImportWorkflow Multi-step import workflow for DCM2NII, NII2BIDS & BIDS2LEGACY.
+%
+% FORMAT: [x] = ExploreASL_ImportWorkflow(x)
+%
+% INPUT:
+%   x             - Struct containing pipeline environment parameters, useful when only initializing ExploreASL/debugging
+%
+% OUTPUT:
+%   x             - Struct containing pipeline environment parameters, useful when only initializing ExploreASL/debugging
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION:    Multi-step import workflow for DCM2NII, NII2BIDS & BIDS2LEGACY.
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% EXAMPLE:        n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    %% Import Workflow
+    
+    % We expect x.DataParPath to be the sourceStructure.json file, so the root directory should be the study directory
+    [x.StudyRoot,~,~] = xASL_fileparts(x.DataParPath);
+    
+    % Check if at least one of the three steps should be performed
+    if sum(x.ImportArray)>0
+        if ~isempty(x.DataParPath)
+            if exist(x.DataParPath,'file')
+                % DICOM TO NII
+                if x.ImportArray(1)==1
+                    ExploreASL_ImportBIDS(x.StudyRoot, x.DataParPath, [], [1 0 0], false, true, false, false, x);
+                end
+                % NII TO BIDS
+                if x.ImportArray(2)==1
+                    ExploreASL_ImportBIDS(x.StudyRoot, x.DataParPath, [], [0 1 0], false, true, false, false, x);
+                end
+                % ANONYMIZE
+                if x.ImportArray(3)==1
+                    ExploreASL_ImportBIDS(x.StudyRoot, x.DataParPath, [], [0 0 1], false, true, false, false, x);
+                end
+                % BIDS TO LEGACY
+                if x.ImportArray(4)==1
+                    x = xASL_Import_BIDS2LEGACY(x);
+                end
+            else
+                warning('ImportArray was set to 1, but the sourceStructure file does not exist');
+            end
+        else
+            warning('ImportArray was set to 1, but there is no DataParPath, Import will not be executed');
+        end
+    end
+    
+    % Reset the import parameter (for the second initialization including the loading of the dataset)
+    x.ImportData = 0;
+    x.ImportArray = [0 0 0 0];
+    
+end
+
+%% xASL_Import_BIDS2LEGACY
+function [x] = xASL_Import_BIDS2LEGACY(x)
```

Comment:
Are you specifying some things that were already specified inside `xASL_bids_BIDS2Legacy`, like the default DataPar fields? I would specify this only once, to facilitate debugging.

Reply:
I'm not sure what you mean. I used the code from `xASL_test_BIDSFlavorsFull` and tried to make it work within this workflow. This needed a few adaptions, which is why we have this subfunction. If there's something specific within the function that you would do differently, please let me know.

Reply:
OK let me have a closer look (or please remind me ;)